[DOCS] Refactor quick start guide and README (#71331)
elastic/elasticsearch: docs/reference/tab-widgets/quick-start-install.asciidoc

// tag::cloud[]
include::{docs-root}/shared/cloud/ess-getting-started.asciidoc[tag=generic]
// end::cloud[]

// tag::self-managed[]

> **Install and run {es} using Docker**

Reviewer: There are also instructions for using Docker compose to spin up multiple ES instances and Kibana so you have the equivalent of a cloud deployment. https://www.elastic.co/guide/en/elastic-stack-get-started/current/get-started-docker.html

Reviewer: Agree that it makes things more complex--mostly mentioned in case you wanted to cross reference it somehow. (Though that's probably not a great idea, as you don't want to send folks off somewhere else.)

ifeval::["{release-state}"=="unreleased"]
NOTE: No Docker image is currently available for {es} {version}.
endif::[]

ifeval::["{release-state}"!="unreleased"]

. Install and start https://www.docker.com/products/docker-desktop[Docker
Desktop].

. Run:
+
[source,sh,subs="attributes"]
----
docker pull {docker-repo}:{version}
docker run --name es01-test -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" {docker-image}
----
endif::[]

**Install and run {kib} using Docker**

> While not required, we recommend installing {kib}. {kib} lets you analyze,
> visualize, and manage {es} data using an intuitive UI.

Reviewer: Maybe instead of leading with "while not required", lead with the motivation?

```suggestion
To analyze, visualize, and manage {es} data using an intuitive UI, install {kib}.
```
